engine/src/main/java/com/google/android/fhir/sync/SyncDownloadExtractor.kt

```diff
@@ -16,17 +16,12 @@
 
 package com.google.android.fhir.sync
 
-import android.content.Context
-import androidx.work.WorkerParameters
-import com.google.android.fhir.resource.TestingUtils
-import org.hl7.fhir.r4.model.ResourceType
+import org.hl7.fhir.r4.model.Resource
 
-class TestFailingSyncWorker(appContext: Context, workerParams: WorkerParameters) :
-  FhirSyncWorker(appContext, workerParams) {
+interface SyncDownloadExtractor {
```

Comment:
just brainstorming, this is optional :)
```suggestion
interface DownloadProcessor {
```


```suggestion
interface DownloadHandler {
```

Reply:
I saw the `FhirSynchronizer` contains a reference to `Uploader`, so I changed the name to `Downloader`. Lmk if that works!

Reply:
i think the uploader does something different, it actually does the uploading!

i think here maybe download processor is better. @aditya-07 even suggested DownloadPrePostProcessor, or DownloadManager?

Reply:
Renamed to `DownloadManager`!